### ISSUE TITLE
[Cover complex] sklearn AgglomerativeClustering: Attribute `affinity` is deprecated

### ISSUE THREAD

**Issue by VincentRouvreau** [GUDHI/gudhi-devel#882]
https://github.com/GUDHI/gudhi-devel/blob/8adafa5cb67bb116854d85a7542046ff47aa83c4/src/python/gudhi/cover_complex.py#L328-L330

Attribute `affinity` was deprecated in version 1.2 and will be removed in 1.4. Use `metric` instead.
cf. https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AgglomerativeClustering.html
